src/nvim/lua/vim.lua

```diff
@@ -1,3 +1,65 @@
+local function system(cmd)
+  local out = vim.api.nvim_call_function('system', { cmd })
+  local err = vim.api.nvim_get_vvar('shell_error')
+  local args_str = vim.api.nvim_call_function('string', { cmd })
+  if 0 ~= err then
+    error('command failed: '..args_str)
+  end
+  return out
+end
+
+local function systemlist(cmd)
```

Comment:
Do we really want to commit a new system interface that mixes `stdout` and `stderr` togheter per default?

Reply:
@bfredl Thanks. I'll keep this internal for now.